clash-ghc/src-ghc/Clash/GHC/ClashFlags.hs

```diff
@@ -1,327 +1,328 @@
 {-|
   Copyright   :  (C) 2015-2016, University of Twente,
                      2016-2017, Myrtle Software Ltd,
-                     2021,      QBayLogic B.V.
+                     2021-2022, QBayLogic B.V.
   License     :  BSD2 (see the file LICENSE)
   Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+
+  Clash command line flag definitions.
 -}
 
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiWayIf #-}
 
 module Clash.GHC.ClashFlags
   ( parseClashFlags
-  , flagsClash
-  )
-where
+  , ghcClashFlags
+  , renderClashFlags
+  ) where
 
+-- ghc
 #if MIN_VERSION_ghc(9,0,0)
-import           GHC.Driver.CmdLine
-import           GHC.Utils.Panic
-import           GHC.Types.SrcLoc
+import GHC.Driver.CmdLine
+  (addErr, errorsToGhcException, processArgs, Err(errMsg), EwM, Flag, Warn)
+import GHC.Utils.Panic (throwGhcExceptionIO)
+import GHC.Types.SrcLoc (Located, unLoc)
 #else
 import           CmdLineParser
-import           Panic
-import           SrcLoc
+  (addErr, errorsToGhcException, processArgs, Err(errMsg), EwM, Flag, Warn)
+import           Panic (throwGhcExceptionIO)
+import           SrcLoc (Located, unLoc)
 #endif
 
+-- base
 import           Control.Monad
-import           Data.Char                      (isSpace)
+import           Data.Char (toLower)
 import           Data.IORef
-import           Data.List                      (dropWhileEnd)
-import           Data.List.Split                (splitOn)
-import qualified Data.Set                       as Set
-import qualified Data.Text                      as Text
-import           Text.Read                      (readMaybe)
+import           Data.List (intercalate)
+import           Text.Read (readMaybe)
+
+-- unordered-containers
+import qualified Data.Set as Set
+import           Data.Set (Set)
```

Comment:
I'm also not a fan of these marker comments for package providence. Especially when it's wrong like this

Reply:
The main reason I'm against it is I don't see what benefit it brings. If you want to see if a package is unused there's `-Wunused-packages`, or `weeder` or any other tool which is going to be quicker and more correct than using these comments. If it's just because there's a lot of imports and it's difficult to read that's more a motivation to split up the module IMO